### PR TITLE
Fix of null column error in geqrf! and gerqf!

### PR DESCRIPTION
**Pull request by andreasvarga**
Addresses [#42762](https://github.com/JuliaLang/LinearAlgebra.jl/issues/882). This is my first PR. I hope it is correct. No tests have been performed.

### CODE REVIEW

**Review by andreasnoack — COMMENTED** [review]
Thanks the PR. It would be good if you could also add a test for the case that used to error. 

Moving `BlasInt` outside `max` will also allow you to use it just once but that isn't a big deal.

**Comment by N5N3** [comment]
`lwork = max(BlasInt(1),BlasInt(real(work[1])))` should be better as there's no need for `max` to handle `NaN` inputs. (Although this won't be the bottleneck)
As for CI, `tester_freebsd64` is known broken for a long time; `tester_linuxaarch64` seems to be related with OpenBLAS itself. So I think you can ignore them here.

As for the test, I think someting like:
```julia
a = rand(2,0), zeros(0)
@test LinearAlgebra.LAPACK.geqrf!(a...) === a
```
would be enough

**Comment by andreasvarga** [comment]
The correction for `gerqf! `was more tricky. Here the issue was that for null column dimension, the required workspace must be at least `M`, the first dimension of the matrix. So, calling the LAPACK routine with the returned "optimal" value of `LWORK = 1`, still leads to error -7. So, I had to modify the workspace adjustement to 
`lwork  = max(m, 1, Int(real(work[1])))`
Probably 
`lwork  = max(m, Int(real(work[1])))`
would also do the job.

File: stdlib/LinearAlgebra/src/lapack.jl [hunk 1]
```diff
@@ -552,7 +552,7 @@ for (gebrd, gelqf, geqlf, geqrf, geqp3, geqrt, geqrt3, gerqf, getrf, elty, relty
                       m, n, A, max(1,stride(A,2)), tau, work, lwork, info)
                 chklapackerror(info[])
                 if i == 1
-                    lwork = BlasInt(real(work[1]))
+                    lwork = max(m, BlasInt(1), BlasInt(real(work[1])))
```
Comment (N5N3) [inline stdlib/LinearAlgebra/src/lapack.jl:555]: If we use 32bit version BLAS on 64bit system,
 `max(m, BlasInt(1), BlasInt(real(work[1])))` will return a `Int64`.
Replace this with `max(BlasInt(m), BlasInt(1), BlasInt(real(work[1])))`?

Comment (andreasvarga) [reply stdlib/LinearAlgebra/src/lapack.jl]: I  am using `max(BlasInt(m),BlasInt(real(work[1]))) ` since the returned _optimal_ value is always at least 1.

**Comment by andreasnoack** [comment]
> The correction for gerqf! was more tricky.

I'm pretty sure that should be considered a but in LAPACK.

**Comment by andreasvarga** [comment]
I opened an issue for LAPACK at [#633](https://github.com/Reference-LAPACK/lapack/issues/633), where I described the problem more in detail.

**Comment by ViralBShah** [comment]
@andreasvarga Could you rebase this to master?

**Comment by andreasvarga** [comment]
This being my first PR, I am afraid I need some guidance what I have exactly to do. Sorry for this!

**Comment by ViralBShah** [comment]
@andreasvarga On the `master` branch do `git pull` to update to the latest julia source code. Then switch to your branch and do `git rebase master`. This will apply your changes on top of the latest julia source (which will basically pull in other fixes and most likely make the failing tests pass).

**Comment by andreasvarga** [comment]
I followed your hints. Here is what I got:

```
$ git pull
remote: Enumerating objects: 1497, done.
remote: Counting objects: 100% (986/986), done.
remote: Compressing objects: 100% (177/177), done.
remote: Total 1497 (delta 833), reused 926 (delta 805), pack-reused 511
Receiving objects: 100% (1497/1497), 1.63 MiB | 3.64 MiB/s, done.
Resolving deltas: 100% (1029/1029), completed with 267 local objects.
From https://github.com/JuliaLang/julia
 + 93e0a8b065...6d9fc140cf aa/libunwind-1.5.0    -> origin/aa/libunwind-1.5.0  (forced update)
 * [new branch]            avi/inlinedoc         -> origin/avi/inlinedoc
   f52c6368e2..cca9bf3530  backports-release-1.6 -> origin/backports-release-1.6
   21a2784994..8597a4b6d3  backports-release-1.7 -> origin/backports-release-1.7
 + 3f15b33c75...fb61f37daa dk/tridiagsolve       -> origin/dk/tridiagsolve  (forced update)
 * [new branch]            jb/bidi               -> origin/jb/bidi
 * [new branch]            jb/faster-resolve-scopes -> origin/jb/faster-resolve-scopes
   2ab960e113..23fcd8437e  jb/stripmeta          -> origin/jb/stripmeta
 * [new branch]            jn/fastlocks          -> origin/jn/fastlocks
 * [new branch]            ksh/chartests         -> origin/ksh/chartests
   c054dbc6d4..7ac710cbf7  master                -> origin/master
   69455be4cd..e990ca91db  oscardssmith-AbstractRNG-docs -> origin/oscardssmith-AbstractRNG-docs
 + e791da70c2...c8f5d11b54 oscardssmith-exhaustive-Float160-tests -> origin/oscardssmith-exhaustive-Float160-tests  (forced update)
 * [new branch]            sds/extended_slurp    -> origin/sds/extended_slurp
 * [new branch]            sds/rest_svec         -> origin/sds/rest_svec
 * [new branch]            sv-keep-hidden        -> origin/sv-keep-hidden
 + 5fedc98405...669ad59ed7 teh/active_project_watchers -> origin/teh/active_project_watchers  (forced update)
 + 30d09a877e...4130432b6c vs/rm-openlibm        -> origin/vs/rm-openlibm  (forced update)
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> av/lapack


$ git rebase master
Current branch av/lapack is up to date.
```

**Comment by ViralBShah** [comment]
Right - also do `git push -f` now to force push your changes to your branch.

**Comment by ViralBShah** [comment]
Actually I am not sure this has fully rebased it to master. I think the steps are:
```
git checkout master
git pull
git checkout av/lapack
git rebase master
git push -f
```

I wonder if things work differently on forked repos.

**Comment by andreasvarga** [comment]
I performed the last step as:
```
$ git remote -v
avarga  git@github.com:andreasvarga/julia.git (fetch)
avarga  git@github.com:andreasvarga/julia.git (push)
origin  https://github.com/JuliaLang/julia.git (fetch)
origin  https://github.com/JuliaLang/julia.git (push)

$ git push -f avarga av/lapack
Enumerating objects: 45, done.
Counting objects: 100% (45/45), done.
Delta compression using up to 16 threads
Compressing objects: 100% (38/38), done.
Writing objects: 100% (38/38), 3.31 KiB | 1.65 MiB/s, done.
Total 38 (delta 29), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (29/29), completed with 6 local objects.
To github.com:andreasvarga/julia.git
 + cdc018b409...d8f2aa7845 av/lapack -> av/lapack (forced update)
```
I hope I did not something wrong.

**Comment by ViralBShah** [comment]
It looks like it happened correctly. Thank you. 

@andreasnoack Are we ok to merge this if CI looks good? I doubt the upstream will do anything anytime soon.

**Review by andreasnoack — APPROVED** [review]

**Pull request merged**